src/python/graplctl/graplctl/aws_cdk_ops.py

```diff
@@ -14,6 +14,24 @@
 LOGGER.addHandler(logging.StreamHandler(stream=sys.stdout))
 
 
+def _run_shell_cmd(
+        cmd: str,
+        cwd: str,
+        stdout: IO[AnyStr],
+        stderr: IO[AnyStr],
+        env: Optional[Dict[str, str]] = None,
```

Comment:
`{}` triggers my "mutable default args" code smell senses, which is why I went for `Optional` here, but in addition to that I did notice some weird behavior in testing. If I don't specify `env` (it defaults to `None` in the `subprocess.run(..)` call) then the underlying shell picks up `PATH` and can locate the `cdk` binary, but if I *do* specify `env` I must include `PATH` in that `env` mapping, otherwise the shell can't find the `cdk` binary. So I suspect specifying an empty `env` would clobber the entire environment, but I haven't tested that.

Reply:
I was able to simply remove the `env` argument entirely.